spark/src/test/scala/org/apache/spark/sql/delta/columnmapping/RemoveColumnMappingRowTrackingSuite.scala

```diff
@@ -21,65 +21,132 @@ import org.apache.spark.sql.delta.DeltaLog
+          checkRowIdsStayTheSameBeforeAndAfterOperation() {
+            assert(!rowTrackingMarkedAsPreservedForCommit(deltaLog) {
+              testRemovingColumnMapping()
+            })
+          }
+        }
+      }
+    }
+  }
+  private def verifyRowIdsStayTheSame(originalRowIds: Array[Row]) = {
+    val newRowIds = spark.read.table(testTableName)
+      .select(logicalColumnName, RowId.QUALIFIED_COLUMN_NAME)
+    checkAnswer(newRowIds, originalRowIds)
+  }
+
+  private def createTestTable(tblProperties: Seq[String] = Seq.empty): Unit = {
+    val tblPropertiesStr = if (tblProperties.nonEmpty) {
+      s"TBLPROPERTIES ${tblProperties.mkString("(", ",", ")")}"
+    } else {
+      ""
+    }
     sql(
-      s"""CREATE TABLE $testTableName
+      s"""
+         |CREATE TABLE $testTableName
          |USING delta
-         |TBLPROPERTIES (
-         |'${DeltaConfigs.COLUMN_MAPPING_MODE.key}' = 'name',
-         |'${DeltaConfigs.ROW_TRACKING_ENABLED.key}' = 'true'
-         |)
+         |$tblPropertiesStr
          |AS SELECT id as $logicalColumnName, id + 1 as $secondColumn
          |  FROM RANGE(0, $totalRows, 1, $numFiles)
          |""".stripMargin)
+  }
 
-    val deltaLog = DeltaLog.forTable(spark, TableIdentifier(tableName = testTableName))
-
-    val snapshot = deltaLog.update()
+  private def checkRowIdsStayTheSameBeforeAndAfterOperation()(operation: => Unit): Unit = {
     val originalDf = spark.read.table(testTableName)
     val originalRowIds = originalDf.select(logicalColumnName, RowId.QUALIFIED_COLUMN_NAME)
       .collect()
-    val originalDomainMetadata = RowTrackingMetadataDomain.fromSnapshot(snapshot).get
 
-    testRemovingColumnMapping()
+    operation
 
     verifyRowIdsStayTheSame(originalRowIds)
-    verifyDomainMetadata(deltaLog.update(), originalDomainMetadata, diffRows = totalRows)
+  }
 
-    // Add back column mapping and remove it again. Row ids should stay the same
-    sql(
-      s"""ALTER TABLE $testTableName
-         |SET TBLPROPERTIES ('${DeltaConfigs.COLUMN_MAPPING_MODE.key}' = 'name')""".stripMargin)
-    // Update a row from each file to force materialize the row ids and metadata.
-    val predicate = s"$logicalColumnName % $rowsPerFile == 0"
-    withSQLConf(UPDATE_USE_PERSISTENT_DELETION_VECTORS.key -> "false") {
-      sql(s"UPDATE $testTableName SET $secondColumn = -1 WHERE $predicate ")
+  private def checkRowIdHighWaterMarkBeforeAndAfterOperation
+      (log: DeltaLog, diffRows: Int)(operation: => Unit): Unit = {
+    def verifyRowIdHighWaterMark(
+        snapshot: Snapshot,
+        originalDomainMetadata: RowTrackingMetadataDomain) = {
+      val newDomainMetadata = RowTrackingMetadataDomain.fromSnapshot(snapshot).get
+      assert(newDomainMetadata.rowIdHighWaterMark ===
+        originalDomainMetadata.rowIdHighWaterMark + diffRows,
+        "Should increase the high watermark by the number of rewritten rows.")
```

Comment:
This inner method is used only once: inline its code for better readability rather than defining a method 

```
  private def checkRowIdHighWaterMarkBeforeAndAfterOperation
      (log: DeltaLog, diffRows: Int)(operation: => Unit): Unit = {
    val originalDomainMetadata = RowTrackingMetadataDomain.fromSnapshot(log.update()).get

    operation

    val newDomainMetadata = RowTrackingMetadataDomain.fromSnapshot(snapshot).get
    assert(newDomainMetadata.rowIdHighWaterMark ===
       originalDomainMetadata.rowIdHighWaterMark + diffRows,
      "Should increase the high watermark by the number of rewritten rows.")
```